pkg/utils/test_util.go

```diff
@@ -32,6 +34,19 @@ var (
 
 	// IgnoreConditionLTTAndMessageFields is a cmpopts.IgnoreFields that ignores the LastTransitionTime and Message fields
 	IgnoreConditionLTTAndMessageFields = cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime", "Message")
+
```

Comment:
please refer to https://github.com/Azure/fleet/pull/1077 for this util

Reply:
I used prometheus built-in Getxxx() functions. Nil condition is handled natively.